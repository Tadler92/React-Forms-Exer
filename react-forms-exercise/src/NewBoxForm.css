form label {
  margin-left: 15px;
}

.NewBoxForm-width-range {
  display: inline-block;
  width: 150px;
}

.NewBoxForm-height-range {
  display: inline-block;
  width: 150px;
}