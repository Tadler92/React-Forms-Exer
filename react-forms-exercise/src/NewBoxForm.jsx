import { useState } from "react";
import './NewBoxForm.css';

const NewBoxForm = ({addBox}) => {
  const INITIAL_STATE = {
    backgroundColor: "",
    width: '',
    height: ''
  }
  const [formData, setFormData] = useState(INITIAL_STATE);

  const handleChange = (e) => {
    const {name, value} = e.target;
    setFormData(formData => ({
      ...formData,
      [name]: value
    }))
  }

  const handleSubmit = (e) => {
    e.preventDefault();
    // addItem(formData.name, formData.qty);
    addBox({...formData});
    setFormData(INITIAL_STATE);
  }

  return (
    <form className="NewBoxForm" onSubmit={handleSubmit}>
      <label htmlFor="backgroundColor">Background Color: </label>
      <input 
        id="backgroundColor"
        type="text" 
        name="backgroundColor" 
        placeholder="Color" 
        value={formData.backgroundColor}
        onChange={handleChange} 
      />

      <label htmlFor="width">Width: </label>
      <div className="NewBoxForm-width-range">
        <p>{formData.width}</p>
        <input 
          id="width"
          type="range" 
          min="10"
          max="200"
          step="10"
          name="width" 
          value={formData.width}
          onChange={handleChange} 
        />
      </div>

      <label htmlFor="height">Height: </label>
      <div className="NewBoxForm-width-range">
        <p>{formData.height}</p>
        <input 
          id="height"
          type="range" 
          min="10"
          max="200"
          step="10"
          name="height" 
          value={formData.height}
          onChange={handleChange} 
        />
      </div>

      <button>Add Box</button>
    </form>
  )
}



export default NewBoxForm;