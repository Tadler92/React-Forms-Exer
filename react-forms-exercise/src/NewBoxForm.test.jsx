import {render, fireEvent, waitFor} from "@testing-library/react";
import NewBoxForm from "./NewBoxForm";
import { expect, it } from "vitest";


describe("NewBoxForm Component", () => {
  it("renders without crashing", function () {
    render(<NewBoxForm />);
  });
  
  
  it("matches snapshot", function () {
    const {asFragment} = render(<NewBoxForm />);
    expect(asFragment()).toMatchSnapshot();
  });
});