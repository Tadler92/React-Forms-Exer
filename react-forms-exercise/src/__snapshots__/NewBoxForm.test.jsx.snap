// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NewBoxForm Component > matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="NewBoxForm"
  >
    <label
      for="backgroundColor"
    >
      Background Color: 
    </label>
    <input
      id="backgroundColor"
      name="backgroundColor"
      placeholder="Color"
      type="text"
      value=""
    />
    <label
      for="width"
    >
      Width: 
    </label>
    <div
      class="NewBoxForm-width-range"
    >
      <p />
      <input
        id="width"
        max="200"
        min="10"
        name="width"
        step="10"
        type="range"
        value=""
      />
    </div>
    <label
      for="height"
    >
      Height: 
    </label>
    <div
      class="NewBoxForm-width-range"
    >
      <p />
      <input
        id="height"
        max="200"
        min="10"
        name="height"
        step="10"
        type="range"
        value=""
      />
    </div>
    <button>
      Add Box
    </button>
  </form>
</DocumentFragment>
`;
